src/cmake-tools.ts

```diff
@@ -244,7 +244,7 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
         try {
           this._statusMessage.set(localize('reloading.status', 'Reloading...'));
           await drv.setBuildPreset(expandedBuildPreset);
-          await this.reRegisterTaskProviderforNewTarget(drv, undefined);
+          //this.updateDefaultTargetInTaskProvider(drv, undefined);
```

Comment:
Uncomment this.  You also shouldn't need to pass "undefined" for the second parameter if it has correct type set up in the signature.